ADR/ADR013-use-paper-trail-gem-for-auditing-and-record-backups.md

```diff
@@ -0,0 +1,51 @@
+# ADR013: Add PaperTrail gem for auditing and record backup
+
+Date: 2023-02-27
+
+## Status
+
+Accepted
+
+## Context
+
+We need way to audit changes to database records to identify what has changed and who made those changes. This information
+will be surfaced to users later on when we come to versioning. This is still valuable to record now for any support tickets
+that come in and require us to understand the form/pages history.
+
+There is also a secondary need be able to restore individual records rather than having to restore from database backups.
+
+## Decision
+
+Add [PaperTrail](https://github.com/paper-trail-gem/paper_trail) as part of draft/live work so we can start recording 
+changes to Forms/Pages but also from any other tables we might want to monitor.
+
+## Consequences
+
+We will not be able to identify what has changed and when without having to go back to splunk server logs and trawl 
+through them in the hopes of being able to spot the changes. We also need this information to surface back to the user.
+
+By setting it up now, we can double check the data that's being collected and tweak it before surfacing it to our users. 
+If there are any issues we can reconfig or completely drop and restart this data collection without any questions from our users.
+
+We already know that we will need to do some additional work to identify who made the changes as we don't record that now as 
+forms-api does not have this data. We will need to pass it in from forms-admin. If we do setup PaperTrail now, when we 
+come to do User Management, we can set up the extra details as part of that work.
+
+### Positives
+
+- We can identify exactly what has changed and when it was changed.
+- We can restore from a specific version of Form/Page
```

Comment:
Would this be done via an active record migration? If not what is the proposed method?

Reply:
It could be a migration or we could use rails console to restore it.

As mentioned at this morning standup, we could create a UI/param endpoint that would rollback a version but until we have our first request we don't really know what method we will end up with. The point is that either way it is possible to restore it. I'll add some more content to the ADR about it